OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -270,14 +236,17 @@ public CPos GetRandomBaseCenter()
 		readonly Predicate<Actor> unitCannotBeOrdered;
 
 		CPos initialBaseCenter;
+		CPos defenseCenter;
 		PowerManager playerPower;
 		PlayerResources playerResource;
 		int ticks;
 
 		BitArray resourceTypeIndices;
 
-		AIHarvesterManager harvManager;
-		AISupportPowerManager supportPowerManager;
+		AIHarvesterModule harvModule;
```

Comment:
Extracting these to modules but then keeping the module types hardcoded hides a lot of the potential benefits here.  Would it be feasible to have a single generic `IAIModule` interface and then define a `Dictionary<string, string>` in the `HackyAIInfo` for condition: enable module name?  Or are we still relying too much on calling into the internal details of the classes?  If we are, then please lets use interfaces for each type.

Reply:
> Or are we still relying too much on calling into the internal details of the classes?

I'm afraid so, but I'll check if I can reduce that further.